File: backend/common/src/test/java/com/webchat/platformapi/file/FileKindTest.java
package com.webchat.platformapi.file;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class FileKindTest {

    @Test
    void fromMimeTypeClassifiesCommonTypes() {
        assertEquals(FileKind.IMAGE, FileKind.fromMimeType("image/png"));
        assertEquals(FileKind.AUDIO, FileKind.fromMimeType("audio/mpeg"));
        assertEquals(FileKind.VIDEO, FileKind.fromMimeType("video/mp4"));
        assertEquals(FileKind.DOCUMENT, FileKind.fromMimeType("application/pdf"));
        assertEquals(FileKind.DOCUMENT, FileKind.fromMimeType("text/plain"));
        assertEquals(FileKind.DOCUMENT, FileKind.fromMimeType("text/csv; charset=utf-8"));
        assertEquals(FileKind.DOCUMENT, FileKind.fromMimeType("application/json"));
        assertEquals(FileKind.DOCUMENT, FileKind.fromMimeType("application/vnd.openxmlformats-officedocument.presentationml.presentation"));
        assertEquals(FileKind.OTHER, FileKind.fromMimeType("application/octet-stream"));
        assertEquals(FileKind.OTHER, FileKind.fromMimeType(null));
    }

    @Test
    void fallsBackToFilenameWhenMimeTypeIsGeneric() {
        assertEquals(
                "application/vnd.openxmlformats-officedocument.presentationml.presentation",
                FileKind.normalizeMimeType("application/octet-stream", "deck.pptx")
        );
        assertEquals(FileKind.DOCUMENT, FileKind.fromMimeTypeOrName("application/octet-stream", "report.csv"));
        assertEquals(FileKind.IMAGE, FileKind.fromMimeTypeOrName("", "diagram.svg"));
    }
}

File: uni-app/src/utils/attachments/fileKind.test.ts
import { describe, expect, it } from 'vitest'
import { inferFileExtension, inferFileKind, inferFileKindFromMime, inferFileKindFromName } from './fileKind'

describe('attachments fileKind', () => {
  it('infers extension', () => {
    expect(inferFileExtension('a.PDF')).toBe('pdf')
    expect(inferFileExtension('noext')).toBe('')
    expect(inferFileExtension('')).toBe('')
  })

  it('infers kind from mime', () => {
    expect(inferFileKindFromMime('image/png')).toBe('image')
    expect(inferFileKindFromMime('image/svg+xml')).toBe('image')
    expect(inferFileKindFromMime('audio/mpeg')).toBe('audio')
    expect(inferFileKindFromMime('video/mp4')).toBe('video')
    expect(inferFileKindFromMime('application/pdf')).toBe('document')
    expect(inferFileKindFromMime('application/vnd.openxmlformats-officedocument.presentationml.presentation')).toBe('document')
    expect(inferFileKindFromMime('application/json')).toBe('document')
    expect(inferFileKindFromMime('text/csv')).toBe('document')
    expect(inferFileKindFromMime('application/octet-stream')).toBe('other')
  })

  it('infers kind from name', () => {
    expect(inferFileKindFromName('x.png')).toBe('image')
    expect(inferFileKindFromName('x.mp3')).toBe('audio')
    expect(inferFileKindFromName('x.mp4')).toBe('video')
    expect(inferFileKindFromName('x.pdf')).toBe('document')
    expect(inferFileKindFromName('x.pptx')).toBe('document')
    expect(inferFileKindFromName('x.csv')).toBe('document')
    expect(inferFileKindFromName('x.markdown')).toBe('document')
    expect(inferFileKindFromName('x.html')).toBe('document')
    expect(inferFileKindFromName('x.unknown')).toBe('other')
  })

  it('prefers mime, falls back to name', () => {
    expect(inferFileKind({ mimeType: 'image/jpeg', filename: 'x.pdf' })).toBe('image')
    expect(inferFileKind({ mimeType: '', filename: 'x.pdf' })).toBe('document')
  })
})

